storage/allocator.go

```diff
@@ -169,8 +169,10 @@ func (a *allocator) AllocateTarget(required proto.Attributes, existing []proto.R
 
 func (a *allocator) allocateTargetInternal(required proto.Attributes, existing []proto.Replica,
 	relaxConstraints bool, filter func(*proto.StoreDescriptor, *stat, *stat) bool) (*proto.StoreDescriptor, error) {
-	attrs := append([]string(nil), required.Attrs...)
-	for {
+	// Because more redundancy is better than less, if relaxConstraints, the
+	// matching here is lenient, and tries to find a target by relaxing an
+	// attribute constraint, from last attribute to first.
+	for attrs := append([]string(nil), required.Attrs...); ; attrs = attrs[:len(attrs)-1] {
```

Comment:
I find this less clear than the old version; I think the three-argument `for` statement only adds clarity when the second argument is used.